### ISSUE TITLE
Job id unknown

### ISSUE THREAD

**Issue by metabba** [Polarion/com.polarion.alm.extensions.codereview#20]
Hello ,

I'm struggling to get the Job working, I m getting an GenericJobException and I tried also another JOB "overdue.job example" it's always the same Exception generating. can you please tell what im missing  ?

2020-05-29 20:17:31,447 [ajp-nio-8889-exec-5 | u:admin] ERROR com.polarion.portal.monitor.internal.server.jobs.JobsDataProvider  - Unknown job codereview.assigner
com.polarion.platform.jobs.GenericJobException: Unknown job codereview.assigner
	at com.polarion.platform.jobs.internal.service.scheduler.JobSchedulerService.scheduleImmediatelyLocally(JobSchedulerService.java:376)
	at com.polarion.platform.jobs.internal.service.scheduler.JobSchedulerService.scheduleImmediately(JobSchedulerService.java:341)
	at $IJobSchedulerService_17261a5df9f.scheduleImmediately($IJobSchedulerService_17261a5df9f.java)
	at com.polarion.portal.monitor.internal.server.jobs.JobsDataProvider.sheduleNow(JobsDataProvider.java:217)
	at com.polarion.alm.ui.server.jobs.JobsServiceImpl$4.executeImpl(JobsServiceImpl.java:66)
	at com.polarion.alm.ui.server.jobs.JobsServiceImpl$4.executeImpl(JobsServiceImpl.java:1)
	at com.polarion.portal.server.TransactionalExecuter$1.runWEx(TransactionalExecuter.java:28)
	at com.polarion.core.util.RunnableWEx.runWRet(RunnableWEx.java:61)
	at com.polarion.portal.server.TransactionalExecuter.executeImpl(TransactionalExecuter.java:76)
	at com.polarion.portal.server.TransactionalExecuter.execute(TransactionalExecuter.java:55)
	at com.polarion.portal.server.TransactionalExecuter.execute(TransactionalExecuter.java:25)
	at com.polarion.alm.ui.server.jobs.JobsServiceImpl.sheduleNow(JobsServiceImpl.java:69)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.polarion.alm.ui.server.rpc.RPCServlet.invokeMethod(RPCServlet.java:96)
	at com.polarion.alm.ui.server.rpc.RPCServlet.lambda$0(RPCServlet.java:81)
	at com.sun.proxy.$Proxy101.sheduleNow(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.google.gwt.user.server.rpc.RPC.invokeAndEncodeResponse(RPC.java:587)
	at com.polarion.alm.ui.server.rpc.RPCServlet$Delegate.processCall(RPCServlet.java:238)
	at com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:303)
	at com.polarion.alm.ui.server.rpc.RPCServlet$Delegate.executeSuperProcessCall(RPCServlet.java:274)
	at com.polarion.alm.ui.server.rpc.RPCServlet$Delegate.lambda$0(RPCServlet.java:214)
	at com.polarion.portal.shared.navigation.CollectionAwareRPC.processCallOnServer(CollectionAwareRPC.java:48)
	at com.polarion.alm.ui.server.rpc.RPCServlet$Delegate.processCall(RPCServlet.java:208)
	at com.google.gwt.user.server.rpc.RemoteServiceServlet.processPost(RemoteServiceServlet.java:373)
	at com.google.gwt.user.server.rpc.AbstractRemoteServiceServlet.doPost(AbstractRemoteServiceServlet.java:62)
	at com.polarion.alm.ui.server.rpc.RPCServlet.doPost(RPCServlet.java:122)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:660)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at com.polarion.alm.ui.server.ExpiresFilterForGWT.doFilter(ExpiresFilterForGWT.java:41)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at com.polarion.alm.server.util.VelocityInstanceFilter.doFilter(VelocityInstanceFilter.java:64)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at com.polarion.portal.tomcat.servlets.DoAsFilter.doFilterAndShowUri(DoAsFilter.java:141)
	at com.polarion.portal.tomcat.servlets.DoAsFilter.access$0(DoAsFilter.java:136)
	at com.polarion.portal.tomcat.servlets.DoAsFilter$1.run(DoAsFilter.java:115)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/javax.security.auth.Subject.doAs(Subject.java:423)
	at com.polarion.platform.internal.security.SubjectNDC.doAs(SubjectNDC.java:69)
	at com.polarion.portal.tomcat.servlets.DoAsFilter.doFilterImpl(DoAsFilter.java:112)
	at com.polarion.portal.tomcat.servlets.DoAsFilter.doFilter(DoAsFilter.java:71)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at com.polarion.portal.tomcat.servlets.SecurityCheckFilter.doFilter(SecurityCheckFilter.java:46)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:607)
	at com.polarion.platform.security.auth.PolarionAuthenticator.invokeInternal(PolarionAuthenticator.java:229)
	at com.polarion.platform.security.auth.PolarionAuthenticator.invoke(PolarionAuthenticator.java:223)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
	at org.apache.catalina.authenticator.SingleSignOn.invoke(SingleSignOn.java:291)
	at com.polarion.platform.session.PolarionLocalSingleSignOn.invoke(PolarionLocalSingleSignOn.java:176)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
	at org.apache.coyote.ajp.AjpProcessor.service(AjpProcessor.java:394)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:770)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1415)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:834)

Thanks in advance

**Comment by srcx** [comment]
Hello. The fact that you can't seem to get SDK job example working points at some broader problem. Would you be able to contact Siemens support? Thank you.

**Comment by BenBaccar** [comment]
Hello Stepan,

thanks for your reply, well I have succeeded to get "overdue.job" and "codereview.checker" working but I have tried to deploy my test Plugin and Im getting another Exception created to hivemind file (see -attached Files), I thank you in advance for your help.


2020-06-01 17:52:37,082 [main] FATAL org.apache.hivemind.schema.rules.ReadAttributeRule  - Error at bundleresource://160.fwk1926096844/META-INF/hivemodule.xml, line 11, column 67: Unable to process attribute factory (of element jobUnitFactory): Unable to find interface com.polarion.platform.jobs.IJobUnitFactory (for service tn.foo.test.emptyJobUnitFactory).
org.apache.hivemind.ApplicationRuntimeException: Unable to find interface com.polarion.platform.jobs.IJobUnitFactory (for service tn.foo.test.emptyJobUnitFactory). [bundleresource://160.fwk1926096844/META-INF/hivemodule.xml, line 4, column 97]



[Exceptions.txt](https://github.com/Polarion/com.polarion.alm.extensions.codereview/files/4712527/Exceptions.txt)
[Exceptions.txt](https://github.com/Polarion/com.polarion.alm.extensions.codereview/files/4712534/Exceptions.txt)
[src.zip](https://github.com/Polarion/com.polarion.alm.extensions.codereview/files/4712535/src.zip)

thanks
